modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java

```diff
@@ -322,12 +332,16 @@ protected boolean connectCamera(int width, int height) {
                     mCaptureSession.close();
                     mCaptureSession = null;
                 }
-                createCameraPreviewSession();
```

Comment:
There is already a createCameraPreviewSession call in onOpened camera callback.

Reply:
Internal native cv::Mat works as smart pointer. Even, if the creator removed Java object the natve object is not removed and hold by this reference.